app/beacon.py

```diff
@@ -61,156 +58,48 @@ def inner(*args, **kwargs):
-    api_beacon_head_actual = 'eth/v1/beacon/headers/head'
-    api_get_balances = 'eth/v1/beacon/states/{}/validators'
-    api_get_slot = 'eth/v1/beacon/states/{}/root'
+    api_get_validators = 'eth/v1/beacon/states/{}/validators'
 
     def __init__(self, url, slots_per_epoch):
         self.url = url
         self.slots_per_epoch = slots_per_epoch
-        self.version = session.get(urljoin(url, self.api_version), timeout=DEFAULT_TIMEOUT).json()
 
     @proxy_connect_timeout_exception
     def get_finalized_epoch(self):
         response = session.get(urljoin(self.url, self.api_beacon_head_finality_checkpoints), timeout=DEFAULT_TIMEOUT)
         return int(response.json()['data']['finalized']['epoch'])
 
     @proxy_connect_timeout_exception
-    def get_genesis(self):
-        response = session.get(urljoin(self.url, self.api_genesis), timeout=DEFAULT_TIMEOUT)
-        return int(response.json()['data']['genesis_time'])
-
-    @proxy_connect_timeout_exception
-    def get_actual_slot(self):
-        actual_slots = {}
-        response = session.get(urljoin(self.url, self.api_beacon_head_actual), timeout=DEFAULT_TIMEOUT).json()
-        actual_slots['actual_slot'] = int(response['data']['header']['message']['slot'])
-        response = session.get(urljoin(self.url, self.api_beacon_head_finalized), timeout=DEFAULT_TIMEOUT).json()
-        actual_slots['finalized_slot'] = int(response['data']['header']['message']['slot'])
-        return actual_slots
-
-    @proxy_connect_timeout_exception
-    def _convert_key_list_to_str_set(self, key_list):
-        pubkeys = set()
-        for key in key_list:
-            pubkeys.add('0x' + binascii.hexlify(key).decode())
-
-        return pubkeys
-
-    @proxy_connect_timeout_exception
-    def get_balances(self, slot, key_list):
-        pubkeys = self._convert_key_list_to_str_set(key_list)
-
-        logging.info('Fetching validators from Beacon node...')
-        balances_url = self.api_get_balances.format(slot)
-        logging.info(f'using url "{balances_url}"')
-        url = urljoin(self.url, balances_url)
-        response_json = session.get(url, timeout=LONG_TIMEOUT).json()
+    def get_balances(self, slot, keys_list) -> Tuple[int, int, int]:
+        all_validators = self._fetch_balances(slot, keys_list)
         logging.info(f'Validator balances on beacon for slot: {slot}')
 
-        found_on_beacon_pubkeys = 0
-        total_balance = 0
-        active_validators_balance = 0
+        validator_pub_keys = self._from_bytes_to_pub_keys(keys_list)
 
-        for validator in response_json['data']:
-            pubkey = validator['validator']['pubkey']
-            # Log all validators along with balance
-            if pubkey in pubkeys:
-                validator_balance = int(validator['balance'])
-                total_balance += validator_balance
-                found_on_beacon_pubkeys += 1
+        df = pd.DataFrame.from_records([{**v, **v['validator']} for v in all_validators])
```

Comment:
I don't really like that we're adding a new dependency here. The logic here wasn't complicated, I'm not sure if the extra library made the code much better, but we got an extra point of failure and potential vulnerability. We also need to retest this part of the logic, it seems to me that refactoring in this place is not much needed now